pkg/restapi/vc/operation/operations.go

```diff
@@ -73,19 +80,27 @@ type vcStatusManager interface {
 // Client interface to interact with edv client
 type Client interface {
 	CreateDataVault(config *operation.DataVaultConfiguration) (string, error)
-	CreateDocument(vaultID string, document *operation.StructuredDocument) (string, error)
-	ReadDocument(vaultID, docID string) ([]byte, error)
+	CreateDocument(vaultID string, document *operation.EncryptedDocument) (string, error)
+	ReadDocument(vaultID, docID string) (*operation.EncryptedDocument, error)
+}
+
+type cryptoProvider struct {
+	crypto legacykms.KeyManager
```

Comment:
do not confuse crypto with kms.. they are two different concepts.. KeyManager is a kms, this should be a kmsProvider with a kms field instead of cryptoProvider with crypto field

Reply:
Gotcha. I've updated the code!